src/main/scala/ai/privado/languageEngine/go/passes/SQLQueryParser.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ *
+ */
+
+package ai.privado.languageEngine.go.passes
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.Constants
+import ai.privado.model.sql.{SQLColumn, SQLQuery}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import ai.privado.utility.{SQLNodeBuilder, Utilities, SQLParser as UtilitySQLParser}
+import better.files.*
+import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.generated.nodes.{
+  Literal,
+  NewFile,
+  NewSqlColumnNode,
+  NewSqlQueryNode,
+  NewSqlTableNode
+}
+import io.shiftleft.codepropertygraph.generated.{Cpg, EdgeTypes, Operators}
+import org.slf4j.LoggerFactory
+import overflowdb.{BatchedUpdate, NodeOrDetachedNode}
+import io.shiftleft.semanticcpg.language.*
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class SQLQueryParser(cpg: Cpg) extends PrivadoParallelCpgPass[Literal](cpg) {
```

Comment:
@hiteshbedre Is this a copied file, I see nearly same implementation already done for JS/Python, There I believe the only difference is we are not creating SQL nodes. We can reimplement it more like OOPS and avoid repeatation for future as well